watertap/unit_models/zero_order/tests/test_suboxic_activated_sludge_process_zo.py

```diff
@@ -0,0 +1,223 @@
+        )
+        assert pytest.approx(10.1864, rel=1e-5) == value(
+            model.fs.unit.properties_treated[0].conc_mass_comp["tkn"]
+        )
+        assert pytest.approx(3.6712e-3, rel=1e-5) == value(
+            model.fs.unit.properties_byproduct[0].flow_vol
+        )
+        assert pytest.approx(268.332, rel=1e-5) == value(
+            model.fs.unit.properties_byproduct[0].conc_mass_comp["tss"]
+        )
+        assert pytest.approx(240.957, rel=1e-5) == value(
+            model.fs.unit.properties_byproduct[0].conc_mass_comp["tkn"]
+        )
+        assert pytest.approx(27, abs=1e-5) == value(model.fs.unit.electricity[0])
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_conservation(self, model):
+        for j in model.fs.params.component_list:
+            assert 1e-6 >= abs(
+                value(
+                    model.fs.unit.inlet.flow_mass_comp[0, j]
+                    - model.fs.unit.treated.flow_mass_comp[0, j]
+                    - model.fs.unit.byproduct.flow_mass_comp[0, j]
+                )
+            )
+
+    @pytest.mark.component
+    def test_report(self, model):
+        model.fs.unit.report()
+
+
+def test_costing():
+    m = ConcreteModel()
+    m.db = Database()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    m.fs.params = WaterParameterBlock(
+        default={"solute_list": ["bod", "tss", "tkn", "phosphorus"]}
+    )
+
+    m.fs.costing = ZeroOrderCosting()
+
+    m.fs.unit1 = SuboxicASMZO(
+        default={"property_package": m.fs.params, "database": m.db}
+    )
+
+    m.fs.unit1.inlet.flow_mass_comp[0, "H2O"].fix(10)
+    m.fs.unit1.inlet.flow_mass_comp[0, "bod"].fix(1)
+    m.fs.unit1.inlet.flow_mass_comp[0, "tss"].fix(1)
+    m.fs.unit1.inlet.flow_mass_comp[0, "tkn"].fix(1)
+    m.fs.unit1.inlet.flow_mass_comp[0, "phosphorus"].fix(1)
+    m.fs.unit1.load_parameters_from_database(use_default_removal=True)
+    assert degrees_of_freedom(m.fs.unit1) == 0
+
+    m.fs.unit1.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.costing}
+    )
+
+    assert isinstance(m.fs.costing.suboxic_activated_sludge_process, Block)
+    assert isinstance(m.fs.costing.suboxic_activated_sludge_process.mixer_cost, Var)
+    assert isinstance(m.fs.costing.suboxic_activated_sludge_process.blower_cost, Var)
+    assert isinstance(m.fs.costing.suboxic_activated_sludge_process.MLR_pump_cost, Var)
+    assert isinstance(m.fs.costing.suboxic_activated_sludge_process.RAS_pump_cost, Var)
+    assert isinstance(
+        m.fs.costing.suboxic_activated_sludge_process.automated_valve_cost, Var
+    )
+    assert isinstance(
+        m.fs.costing.suboxic_activated_sludge_process.advanced_predicted_control_cost,
+        Var,
+    )
+    assert isinstance(
+        m.fs.costing.suboxic_activated_sludge_process.ammonia_probe_cost, Var
+    )
```

Comment:
Is this everything? The list looked longer in the cost method.

Reply:
Added all other variables in the test.